lib/pbench/server/cache_manager.py

```diff
@@ -227,9 +227,12 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
         return cls(destination, controller)
 
     @staticmethod
-    def extract(tarball_path: Path, path: str) -> str:
+    def extract(tarball_path: Path, path: str) -> Optional[str]:
         """Extract a file from the tarball and return it as a string
 
+        If we can't open the tarball, fail with a metadata error. However if
+        that succeeds but there's no `metadata.log` inside, succeed.
```

Comment:
The code provides a pretty weak notion of "success"...it returns `None` instead of the hoped-for string....  (You might want to expand on "succeed" here, and the "Returns" description at line 243 could use a tweak as well.)